src/tools/rbd/action/Trash.cc

```diff
@@ -236,8 +236,12 @@ int do_list(librbd::RBD &rbd, librados::IoCtx& io_ctx, bool long_flag,
 
     std::string pool, image, snap, parent;
     r = im.parent_info(&pool, &image, &snap);
-    if (r < 0 && r != -ENOENT)
+    if (r == -ENOENT) {
+      r = 0;
```

Comment:
@dillaman we have `if (r != -ENOENT)` block below.

Reply:
Yeah -- but since it doesn't exit and doesn't clear the error code, that ```-ENOENT``` error code will be returned when the function exits.

Reply:
D'oh -- sorry, never mind